crates/router/src/connector/noon/transformers.rs

```diff
@@ -326,27 +367,34 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for NoonPaymentsRequest {
                 },
             });
 
-        let (subscription, tokenize_c_c) =
-            match item.request.setup_future_usage.is_some().then_some((
+        let (subscription, tokenize_c_c) = match item
+            .router_data
+            .request
+            .setup_future_usage
+            .is_some()
+            .then_some((
                 NoonSubscriptionData {
                     subscription_type: NoonSubscriptionType::Unscheduled,
                     name: name.clone(),
                 },
                 true,
             )) {
-                Some((a, b)) => (Some(a), Some(b)),
-                None => (None, None),
-            };
+            Some((a, b)) => (Some(a), Some(b)),
+            None => (None, None),
+        };
         let order = NoonOrder {
-            amount: conn_utils::to_currency_base_unit(item.request.amount, item.request.currency)?,
+            amount: conn_utils::to_currency_base_unit(
+                item.router_data.request.amount,
+                item.router_data.request.currency,
+            )?,
```

Comment:
Use item.amount, in place of to_currency_base_unit().
As in noon.rs we already call the tryfrom for currency conversion incase of Payment Authorize